### PR TITLE
Made "dynamic sampling page" more readable

### PR DESCRIPTION
**Pull request by antonpirker**
I moved some parts around to have a better flow of the docs. The specification of the DSC is now a table to be easier to read. Removed also the "experimental" note. 

Before:
https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/

Updated:
https://develop-git-antonpirker-dynamic-sampling-context-cleanup.sentry.dev/sdk/performance/dynamic-sampling-context/

### CODE REVIEW

File: src/docs/sdk/performance/dynamic-sampling-context.mdx [hunk 1]
```diff
@@ -1,25 +1,17 @@
...
 - The sampling strategy leverages head-based simple random sampling.
 - Employing sampling rules, for example, based on event parameters, is very complex. Sometimes this even requires users to have an in-depth understanding of the event schema.
 - While writing rules for singular **transactions** is possible, enforcing them on entire **traces** is infeasible.
 
-The solution for these problems is **Dynamic Sampling**.
-Dynamic Sampling allows users to configure **sampling rules** directly in the Sentry interface. Important: Sampling rules may be applied to **entire traces** or to a **single transaction**.
+**Dynamic Sampling** allows users to configure **sampling rules** directly in the Sentry interface. Those sampling rules may be applied to **entire traces** or to a **single transaction**. Dynamic Sampling allows Sentry to automatically adjust the amount of data retained based on the value of the data. For detailed information see [Dynamic Sampling](/dynamic-sampling/).
```
Comment (cleptric) [inline src/docs/sdk/performance/dynamic-sampling-context.mdx:14]: Where can people configure DS in the UI?

File: src/docs/sdk/performance/dynamic-sampling-context.mdx [hunk 2]
```diff
@@ -42,148 +34,140 @@ Multiple different traces can and should have different Dynamic Sampling Context
...
+| `release`                   | string | The release name as specified in client options.                                                                             | `myapp@1.2.3`, `1.2.3`, `2025.4.107` | required                             |
+| `environment`               | string | The environment name as specified in client options.                                                                         | `production`, `staging`              | required                             |
+| `transaction`               | string | The transaction name set on the scope. **Only include** if name has [good quality](#note-on-good-quality-transaction-names). | `/login`, `myApp.myController.login` | required (if known and good quality) |
+| `user_segment` [DEPRECATED] | string | User segment as set by the user with `scope.set_user()`.                                                                     |                                      | deprecated                           |
```
Comment (cleptric) [inline src/docs/sdk/performance/dynamic-sampling-context.mdx:62]: `sampled` is missing, which is set by the head of trace.

File: src/docs/sdk/performance/dynamic-sampling-context.mdx [hunk 3]
```diff
@@ -42,148 +34,140 @@ Multiple different traces can and should have different Dynamic Sampling Context
...
+To align DSC propagation over all our SDKs, we defined a [unified propagation mechanism](#unified-propagation-mechanism) (step-by-step instructions) that all SDK implementations should be able to follow.
 
-#### Currently supported SDKs:
+## DSC Specification
 
-The currently supported SDKs are documented on the [Dynamic Sampling Product docs](https://docs.sentry.io/product/data-management-settings/server-side-sampling/getting-started/#current-limitations). Please update that list once you release an SDK that supports dynamic sampling features. To be considered a supported SDK for dynamic sampling, the SDK must support both creating the DSC at the head of the trace, and propagating the DSC from continuing traces (if possible for that environment).
+All of the attributes in the table below are required (non-optional) in a sense, that when they are known to an SDK at the time an envelope with an event (transaction or error) is sent to Sentry, or at the time a baggage header is propagated, they must also be included in said envelope or baggage. In any case, `trace_id`, `public_key`, and `sample_rate` should always be known to an SDK, so these values are strictly required.
```
Comment (cleptric) [inline src/docs/sdk/performance/dynamic-sampling-context.mdx:49]: Can we clarify required and strictly required please.
Certain values are required only in certain context.

File: src/docs/sdk/performance/dynamic-sampling-context.mdx (hunk 1 above)
Comment (AbhiPrasad) [reply src/docs/sdk/performance/dynamic-sampling-context.mdx:14]: Yeah this is old, we should remove that information

**Review by AbhiPrasad — COMMENTED** [review]
General clean up looks good, we should just update based on Michi's comments.

File: src/docs/sdk/performance/dynamic-sampling-context.mdx (hunk 1 above)
Comment (antonpirker) [reply src/docs/sdk/performance/dynamic-sampling-context.mdx:14]: Ok, will remove

Comment (antonpirker) [reply src/docs/sdk/performance/dynamic-sampling-context.mdx:14]: Ok, will remove

File: src/docs/sdk/performance/dynamic-sampling-context.mdx (hunk 3 above)
Comment (antonpirker) [reply src/docs/sdk/performance/dynamic-sampling-context.mdx:49]: added a sentence below explaining it.

File: src/docs/sdk/performance/dynamic-sampling-context.mdx (hunk 2 above)
Comment (antonpirker) [reply src/docs/sdk/performance/dynamic-sampling-context.mdx:62]: added it

**Pull request merged**